support/metrics/metricsTracker.go

```diff
@@ -286,3 +350,36 @@ func (mt *MetricsTracker) sendEvent(eventType string, eventProps interface{}) er
 	}
 	return nil
 }
+
+func mergeEventProps(commonProps, eventProps map[string]interface{}) (map[string]interface{}, error) {
+	m := make(map[string]interface{})
+	for k, v := range commonProps {
+		m[k] = v
+	}
+
+	for k, v := range eventProps {
+		m[k] = v
+	}
+
+	return m, nil
+}
+
+func toMapStringInterface(v interface{}) (map[string]interface{}, error) {
+	b, e := json.Marshal(v)
+	if e != nil {
+		return nil, fmt.Errorf("could not marshal interface to json: %s", e)
+	}
+
+	var i interface{}
+	e = json.Unmarshal(b, &i)
+	if e != nil {
+		return nil, fmt.Errorf("could not unmarshal json to interface: %s", e)
+	}
+
+	m, ok := i.(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("could not create map[string]interface{}")
```

Comment:
we should also specify the type here in the error message:
`fmt.Errorf("could not create map[string]interface{}, underlying type was '%T'", i)`

not applicable if using `map[string]interface{}` works above